pennylane/gradients/parameter_shift.py

```diff
@@ -376,18 +376,17 @@ def processing_fn(results):
         ragged = getattr(results[0], "dtype", None) is np.dtype("object")
 
         mask = []
-        for m, r in zip(var_mask, results[0]):
+        for m, r in zip(var_mask, qml.math.atleast_1d(results[0])):
             array_func = np.ones if m else np.zeros
             shape = qml.math.shape(r)
-            shape = (1,) if shape == tuple() else shape
             mask.append(array_func(shape, dtype=bool))
 
         if ragged:
             res = qml.math.hstack(res)
             mask = qml.math.hstack(mask)
 
-        mask = qml.math.convert_like(qml.math.reshape(mask, [-1, 1]), res)
         f0 = qml.math.expand_dims(res, -1)
+        mask = qml.math.convert_like(qml.math.reshape(mask, qml.math.shape(f0)), res)

```

Comment:
was this a bug?

Reply:
The deleted reshape was forcing the mask to always be 2d, so there would be an extra dimension in the output even when the data itself was 1d. This way the mask adapts to the shape of the data.